packages/loader/container-loader/src/container.ts

```diff
@@ -649,7 +647,11 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             connectionState: ConnectionState[this.connectionState],
         });
 
-        if (reconnect) {
+        // If container state is not attached, then don't connect to delta stream. Also don't set the
+        // manual reconnection flag to true as we haven't made the initial connection yet.
+        if (reconnect && this._attachState === AttachState.Attached) {
+            // Attached container must have been resumed.
+            assert(this.resumedOpProcessingAfterLoad);
```

Comment:
I think we can convert this assert into condition to add into IF statement above.

Reply:
But I won't call this a condition, this should be an assert because an attached container must have been resumed.

Reply:
Not necessarily, It's similar to detached container - the call only sets the policy (of keeping or dropping reconnects), the impact of that policy happens only when container is attached and resumed.